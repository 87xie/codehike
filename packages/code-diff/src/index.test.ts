import { codeDiff } from "./index"

const prevCode = `
log(a)
x = "x"
`.trim()
const nextCode = `
log(a)
x = "x"
y = "y"
`.trim()

test("basic", () => {
  expect(
    codeDiff({
      prevCode,
      nextCode,
      lang: "js",
    })
  ).toMatchSnapshot()
})

const p = [1, 2, 3, 4]
const n = [1, 2, 3, 3.5, 4]
const m = { 1: "A" }
