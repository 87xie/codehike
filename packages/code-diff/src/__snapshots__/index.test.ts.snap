// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
Object {
  "codeMap": Object {
    "1": Array [
      Array [
        "log",
        "function",
      ],
      Array [
        "(",
        "punctuation",
      ],
      Array [
        "a",
        "plain",
      ],
      Array [
        ")",
        "punctuation",
      ],
    ],
    "2": Array [
      Array [
        "x ",
        "plain",
      ],
      Array [
        "=",
        "operator",
      ],
      Array [
        " ",
        "plain",
      ],
      Array [
        "\\"x\\"",
        "string",
      ],
    ],
    "2.5": Array [
      Array [
        "y ",
        "plain",
      ],
      Array [
        "=",
        "operator",
      ],
      Array [
        " ",
        "plain",
      ],
      Array [
        "\\"y\\"",
        "string",
      ],
    ],
    "3": Array [],
  },
  "nextKeys": Array [
    1,
    2,
    2.5,
  ],
  "prevKeys": Array [
    1,
    2,
    3,
  ],
}
`;
